pkg/util/wait/backoff.go

```diff
@@ -0,0 +1,76 @@
+package wait
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/utils/clock"
+	"k8s.io/utils/ptr"
+)
+
+// UntilWithBackoff runs f in a loop until context indicates finished. It
+// applies backoff depending on the SpeedSignal f returns.  Backoff increases
+// exponentially, ranging from 1ms to 100ms.
+func UntilWithBackoff(ctx context.Context, f func(context.Context) SpeedSignal) {
+	// create and drain timer, allowing reuse of same timer via timer.Reset
+	timer := clock.RealClock{}.NewTimer(0)
+	<-timer.C()
+	untilWithBackoff(ctx, f, timer)
+}
+
+func untilWithBackoff(ctx context.Context, f func(context.Context) SpeedSignal, timer clock.Timer) {
+	mgr := speedyBackoffManager{
+		backingOff:  ptr.To(false),
+		rateLimiter: workqueue.NewItemExponentialFailureRateLimiter(initialBackoff, maxBackoff),
+		timer:       timer,
+	}
+	wait.BackoffUntil(func() {
+		mgr.toggleBackoff(f(ctx))
+	}, mgr, false, ctx.Done())
```

Comment:
```suggestion
	}, &mgr, false, ctx.Done())
```

This should remove the need for a boolean pointer.

BackoffManager is an interface, so it can accept a pointer or a value.